### PR TITLE
stripe

### PR DESCRIPTION
**Pull request by cgardens**
## Testing
* This has been tested pushing the `customers` stream into local csv destination. that's it. need to to do more testing. 

labels: 
* stripe - issue specific to our stripe integration.
* core - issue with the core workings of our app.

## What
* (**stripe**) integration expected to receive `--catalog` worker sends `--properties`
* (**core**) airbyte didn't support all the singer types (object, array)
* (**stripe**) integration did not handle the fact that it wrote stuff other than the catalog json to stdout
* (**core**) singer stripe sets a `type` field in metadata using the same sometimes-an-array that we see with singer type. a little unclear why, but now we treat that field like a singertype. this is because we assumed a table / column structure. we're going to have to go back and take a more careful look this. 
* (**stripe** ?) singer stripe catalog includes key_properties, we didn't have that in our singercatalog. added. this field don't seem to be in the singer [spec](https://github.com/singer-io/getting-started/blob/master/docs/DISCOVERY_MODE.md).
* (**core**) singer stripe throws add a bunch of fields in `SingerColumn` (e.g. `anyOf`). this is related to the bad assumption we made about the shape of schemas. these extra fields is the rest of the jsonschema syntax. again, we'll need to take a closer look at schema in the next iteration.
* (**core**) write a test to make sure that we always have a mapping from all `SingerType`s to all of our `DataType`s
* (**stripe**) write a test to make sure we can deserialize a stripe catalog and schema messages.
* (**core**) stop adding empty collections for fields in json2schema pojos. it meant we were outputting invalid jsonschema which was causing failures. this was causing really hard to debug errors.
  * learned that the stripe tap accepts the catalog.json on a sync and outputs it as the schema message without validation. this meant that when the target tried to validate the message it failed! this was hard to figure out since it was our deserialization protocol that was adding the invalid fields.
* (**core**) todo: address the fact that schema messages get rejected in the sync stream because the validator treats singertype as an object instead of an array. have to look into this next, but we can move forward with a hack that just lets all schema messages through
* (**core**) if singer data type is `anyOf`, etc, treat it as "object" in our schema and treat it as json node when deserializing it.

### CODE REVIEW

File: airbyte-workers/src/test/java/io/airbyte/workers/protocols/singer/SingerCatalogConvertersTest.java [hunk 1]
```diff
@@ -64,6 +72,16 @@ void applySchemaToDiscoveredCatalog() throws IOException {
     assertEquals(expectedCatalog, actualCatalog);
   }
 
+  @Test
```
Comment (cgardens) [inline airbyte-workers/src/test/java/io/airbyte/workers/protocols/singer/SingerCatalogConvertersTest.java:75]: this belongs in the integration module not here.

File: airbyte-singer/build.gradle [hunk 2]
```diff
@@ -36,6 +36,16 @@ task postprocess {
                 writer.writeLine(it)
             }
         }
+        def original2 = new File(project.buildDir, 'generated/src/gen/java/io/airbyte/singer/SingerTableSchema.java')
```
Comment (cgardens) [inline airbyte-singer/build.gradle:39]: this needs to be cleaned up

File: airbyte-workers/src/main/java/io/airbyte/workers/protocols/singer/DefaultSingerStreamFactory.java [hunk 3]
```diff
@@ -75,6 +75,13 @@ public DefaultSingerStreamFactory() {
         .filter(Optional::isPresent)
         .map(Optional::get)
         .filter(j -> {
+          // todo (cgardens) - hack. seems like the validation in the worker does not play well with
```
Comment (cgardens) [inline airbyte-workers/src/main/java/io/airbyte/workers/protocols/singer/DefaultSingerStreamFactory.java:78]: i'm inclined to work on this next as this is a pretty awful hack.

File: airbyte-integrations/singer/stripe/source/src/test-integration/java/io/airbyte/integration_tests/sources/SingerStripeSourceUnitTests.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.singer.SingerCatalog;
+import io.airbyte.singer.SingerMessage;
+import java.io.IOException;
+import org.junit.jupiter.api.Test;
+
+public class SingerStripeSourceUnitTests {
```
Comment (michel-tricot) [inline airbyte-integrations/singer/stripe/source/src/test-integration/java/io/airbyte/integration_tests/sources/SingerStripeSourceUnitTests.java:38]: Let's not call it UnitTests

**Review by michel-tricot — APPROVED** [review]

**Review by jrhizor — APPROVED** [review]

**Pull request merged**